src/combat.cpp

```diff
@@ -776,7 +775,9 @@ void Combat::doTargetCombat(Creature* caster, Creature* target, CombatDamage& da
 		if (g_game.combatBlockHit(damage, caster, target, params.blockedByShield, params.blockedByArmor, params.itemId != 0)) {
 			return;
 		}
-
+		if (casterPlayer) {
```

Comment:
why did you move the code here?

Reply:
I moved it here to be after block check.

Reply:
> why did you move the code here I think you can just check the target variable if it leechs even on no target?

I moved it here to be after block check.